### DISCUSSION TITLE
Issues scanning small QR codes (20x20mm) with high error correction

### DISCUSSION THREAD

**Discussion by TriJoe** [mebjas/html5-qrcode#528] [Q&A]
We're having issues scanning small QR codes. The QR codes are about 20x20mm in size, and use high error correction. We can't change the QR code sizes as the factory had already applied them to a large batch of products. 

The issue is that it takes a while, moving a phone closer and further, changing the angle until it scans, which can take 10 seconds or more to scan. 

Using the camera app on the phone scans instantly. 

We've tested with Android S21 / S20, iPhone 5 and iPhone 13, as well as some cheaper Android phones. All have the same issue. 

Has anyone else had these issues?

**Comment by ROBERT-MCDOWELL** [comment]
"Using the camera app on the phone scans instantl"
you should understand that browser based scanner are based on javascript libraries or the browser API itself which for now is not as efficient as an app if you are not accurately set your javascript code.
I suggest you use zoom constraint to help scanning.

**Reply by TriJoe** [reply]
Thanks, I wanted to point out that it's not a camera / lighting issue. Although the image from the scanning app does look at lot worse than that of the camera, lower resolution and doesn't adjust for lighting (on my S21 anyway). 

I will test with zoom. Thanks.

**Comment by ROBERT-MCDOWELL** [comment]
if the image is blur so you must use the focusmode constraint too

**Reply by TriJoe** [reply]
Getting the phone close enough does make it blurry, and strangely on iPhones, the QR code turns almost black. I will test with this too.
